nltk/stem/lancaster.py

```diff
@@ -192,14 +201,20 @@ def stem(self, word):
         """Stem a word using the Lancaster stemmer.
         """
         # Lower-case the word, since all the rules are lower-cased
-        word = word.lower()
+        word = self.strip_prefix(word).lower() if self._strip_prefix else word.lower()
```

Comment:
- [ ] Surely we should be coercing the word to lowercase *before* the `strip_prefix()` call, not after? Won't this implementation map `KILOGRAM` to `kilogram` but `kilogram` to `gram`?